### PR TITLE
refactor EVO::sendRedirect()

### PR DESCRIPTION
**Pull request by yama**

### CODE REVIEW

**Comment by Pathologic** [comment]
I see no sense in new options. REDIRECT_JS and REDIRECT_META both require the call to be in the right place, so what's the difference between calling sendRedirect with REDIRECT_JS option and writing "<script>window.location = " in the template. $this->getService('ExceptionHandler')->messageQuit('No newline allowed in redirect url.') - maybe throwing exception will be better? And please stop abusing with sprintf.

**Comment by yama** [comment]
@Pathologic 
ok, I have separate commits, so you can cherry-pick.

**Comment by yama** [comment]
(ps)
Hard to explain, but REDIRECT_JS is what we need. If the core does not support it, it will support it independently.

File: core/src/Core.php [hunk 1]
```diff
@@ -409,62 +409,51 @@ public function getMicroTime()
...
-            } else {
-                $currentNumberOfRedirects += 1;
-                if (Str::contains($url, '?')) {
-                    $url .= "&err=$currentNumberOfRedirects";
-                } else {
-                    $url .= "?err=$currentNumberOfRedirects";
-                }
+                exit;
             }
+            $url .= (Str::contains($url, '?') ? '?' : '&') . 'err=' . ($currentNumberOfRedirects + 1);
```
Comment (Pathologic) [inline core/src/Core.php]: Should it be
```
$url .= (Str::contains($url, '?') ? '&' : '?') . 'err=' . ($currentNumberOfRedirects + 1);
```
?

File: core/src/Core.php [hunk 2]
```diff
@@ -432,11 +432,7 @@ public function sendRedirect($url, $count_attempts = 0, $type = 'REDIRECT_HEADER
                 );
                 exit;
             }
-            $url .= sprintf(
-                '%serr=%s',
-                Str::contains($url, '?') ? '?' : '&',
-                ($currentNumberOfRedirects + 1)
-            );
+            $url .= (Str::contains($url, '?') ? '?' : '&') . 'err=' . ($currentNumberOfRedirects + 1);
```
Comment (Pathologic) [inline core/src/Core.php]: The problem is not in lines :smile_cat: If Str::contains($url, '?') is true then & should be added, not ?

**Pull request merged**